src/main/webapp/app/exercises/programming/manage/grading/testcase-analysis/testcase-analysis.component.ts

```diff
@@ -0,0 +1,85 @@
+import { Component, Input, OnInit } from '@angular/core';
+import { ArtemisSharedModule } from 'app/shared/shared.module';
+import { SimplifiedTask, TestcaseAnalysisService } from 'app/exercises/programming/manage/grading/testcase-analysis/testcase-analysis.service';
+import { concatMap, tap } from 'rxjs/operators';
+
+type FeedbackDetail = {
+    count: number;
+    detailText: string;
+    testcase: string;
+    task: number;
+};
+
+@Component({
+    selector: 'jhi-testcase-analysis',
+    templateUrl: './testcase-analysis.component.html',
+    standalone: true,
+    imports: [ArtemisSharedModule],
+    providers: [TestcaseAnalysisService],
+})
+export class TestcaseAnalysisComponent implements OnInit {
+    @Input() exerciseTitle?: string;
+    @Input() exerciseId?: number;
+    resultIds: number[] = [];
+    tasks: SimplifiedTask[] = [];
+    feedback: FeedbackDetail[] = [];
+
+    constructor(private simplifiedProgrammingExerciseTaskService: TestcaseAnalysisService) {}
+
+    ngOnInit(): void {
+        if (this.exerciseId) {
+            this.loadTasks(this.exerciseId)
+                .pipe(concatMap(() => this.loadFeedbackDetails(this.exerciseId!)))
+                .subscribe();
+        }
+    }
+
+    loadTasks(exerciseId: number) {
```

Comment:
Can you add the return type here? `loadTasks` sounds a bit like a `void` method

Reply:
it doe not return void 👍